sklearn/calibration.py

```diff
@@ -390,6 +390,11 @@ def fit(self, X, y, sample_weight=None, **fit_params):
                     "cross-validation but provided less than "
                     f"{n_folds} examples for at least one class."
                 )
+            if isinstance(self.cv, LeaveOneOut):
+                raise ValueError(
+                    "LeaveOneOut cross-validation does not allow"
+                    "all classes to be present in test splits."
```

Comment:
Let's be explicit by asking people to use an alternative cross-validation strategies.

Reply:
Amended. Hopefully not too long now?

Reply:
Nop this is good.